extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/DimensionsAndCollector.java

```diff
@@ -33,7 +37,7 @@ public class DimensionsAndCollector
   private final SimpleCollector collector;
   private final double conversionFactor;
   private final double[] histogramBuckets;
-  private final Stopwatch updateTimer;
+  private final ConcurrentHashMap<List<String>, Stopwatch> labelValuesToStopwatch;
```

Comment:
Did you see any memory pressure with this change? Previously [there](https://github.com/apache/druid/pull/18598#discussion_r2409716522) were some concerns on the memory usage with this approach, so it would be nice if you could share any observations